components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/filter/FilterResult.java

```diff
@@ -0,0 +1,82 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.uuf.internal.filter;
+
+import java.util.Optional;
+
+/**
+ * Bean class for holding the filter result.
+ *
+ * @since 1.0.0
+ */
+public class FilterResult {
+
+    private Integer statusCode;
```

Comment:
int value is by default 0. If the user decides not to set the status code (using `public FilterResult(boolean isContinue)` constructor) then if we get the status code, we will be returning the wrong value 0. Therefore this shouldn't be the primitive type. WDYT?